pkg/runtime/service_controller.go

```diff
@@ -233,6 +234,10 @@ func (c *serviceController) BindControllerManager(mgr ctrlrt.Manager, cfg ackcfg
 		// Run the caches. This will not block as the caches are run in
 		// separate goroutines.
 		cache.Run(clientSet)
+		// Wait for the caches to sync
+		ctx := context.TODO()
```

Comment:
can we do 1 minute for this instead

Reply:
I think indeed this will solve the failing unit tests, but imposes risk of race condition

Reply:
maybe we can configure that specific failing unit test to not use CARM caches? it is only supposed to tests that the controller build happen correctly

Reply:
what's happening here is that: the unit tests attempt to initialize a cache, but `WaitForCachesToSync` requires communication with the api server, which is not running during unit tests. so... the function waits indefinitely an eventually leads to timeouts..